gammapy/scripts/analysis.py

```diff
@@ -179,31 +202,12 @@ def get_observations(self):
         for obs in self.observations.list:
             log.info(obs)
 
-    def reduce(self):
-        """Produce reduced data sets."""
-        if not self._validate_reduction_settings():
-            return False
-        if self.settings["reduction"]["data_reducer"] == "1d":
-            self._spectrum_extraction()
-        elif self.settings["reduction"]["data_reducer"] == "3d":
-            self._create_geometry()
-            maker_params = {}
-            if "offset_max" in self.settings["geometry"]:
-                maker_params = {"offset_max": self.settings["geometry"]["offset_max"]}
-            maker = MapMaker(self.geom, **maker_params)
-            self.maps = maker.run(self.observations)
-            self.images = maker.run_images()
-            table_psf = make_mean_psf(self.observations, self.geom.center_skydir)
-            psf_params = {}
-            if "psf_max_radius" in self.settings["geometry"]:
-                psf_params = {"max_radius": self.settings["geometry"]["psf_max_radius"]}
-            self.psf_kernel = PSFKernel.from_table_psf(
-                table_psf, self.geom, **psf_params
-            )
-
-        else:
-            # TODO raise error?
-            log.info("Data reduction method not available.")
+    def get_residuals(self):
+        log.info("Calculating residuals.")
+        npred = self.datasets.datasets[0].npred()
+        self.residuals = Map.from_geom(self.geom)
```

Comment:
Note that all of our dataset classes already have `.residuals()` method, the you can just re-use. See:
https://docs.gammapy.org/dev/api/gammapy.cube.MapDataset.html#gammapy.cube.MapDataset.residuals
https://docs.gammapy.org/dev/api/gammapy.spectrum.SpectrumDatasetOnOff.html#gammapy.spectrum.SpectrumDatasetOnOff.residuals